cloudinit/sources/DataSourceAzure.py

```diff
@@ -1335,34 +1620,66 @@ def read_azure_ovf(contents):
 
-    cfg['PreprovisionedVm'] = _extract_preprovisioned_vm_setting(dom)
+    preprovisioning_cfg = _get_preprovisioning_cfgs(dom)
+    cfg = util.mergemanydict([cfg, preprovisioning_cfg])
 
     return (md, ud, cfg)
 
 
 @azure_ds_telemetry_reporter
-def _extract_preprovisioned_vm_setting(dom):
-    """Read the preprovision flag from the ovf. It should not
-       exist unless true."""
+def _get_preprovisioning_cfgs(dom):
+    """Read the preprovisioning related flags from ovf and populates a dict
+    with the info."""
+    cfg = {}
+    cfg["PreprovisionedVm"] = False
+    cfg["PreprovisionedVMType"] = None
+
     platform_settings_section = find_child(
         dom.documentElement,
         lambda n: n.localName == "PlatformSettingsSection")
     if not platform_settings_section or len(platform_settings_section) == 0:
         LOG.debug("PlatformSettingsSection not found")
-        return False
+        return cfg
     platform_settings = find_child(
         platform_settings_section[0],
         lambda n: n.localName == "PlatformSettings")
     if not platform_settings or len(platform_settings) == 0:
         LOG.debug("PlatformSettings not found")
-        return False
+        return cfg
+
+    # Read the PreprovisionedVm bool flag. This should be deprecated when the
+    # platform has removed PreprovisionedVm and only surfaces
+    # PreprovisionedVMType.
     preprovisionedVm = find_child(
         platform_settings[0],
         lambda n: n.localName == "PreprovisionedVm")
     if not preprovisionedVm or len(preprovisionedVm) == 0:
         LOG.debug("PreprovisionedVm not found")
-        return False
-    return util.translate_bool(preprovisionedVm[0].firstChild.nodeValue)
+        return cfg
+    cfg["PreprovisionedVm"] = util.translate_bool(
+        preprovisionedVm[0].firstChild.nodeValue)
+    LOG.info("PreprovisionedVm: %s", cfg["PreprovisionedVm"])
+
+    # Read the PreprovisionedVMType value from the ovf. It can be
+    # 'Running' or 'Savable' or not exist. This enum value is intended to
+    # replace PreprovisionedVm bool flag in the long term.
+    # A Running VM is the same as preprovisioned VMs of today. This is
+    # equivalent to having PreprovisionedVm=True.
+    # A Savable VM is one whose nic is hot-detached immediately after it
+    # reports ready the first time to free up the network resources.
+    # Once assigned to customer, the customer-requested nics are
+    # hot-attached to it and reprovision happens like today.
+    preprovisionedVMType = find_child(
+        platform_settings[0],
+        lambda n: n.localName == "PreprovisionedVMType")
+    if (not preprovisionedVMType or len(preprovisionedVMType) == 0 or
+            preprovisionedVMType[0].firstChild is None):
+        LOG.debug("PreprovisionedVMType not found")
+        return cfg
+
+    cfg["PreprovisionedVMType"] = preprovisionedVMType[0].firstChild.nodeValue
+    LOG.info("PreprovisionedVMType: %s", cfg["PreprovisionedVMType"])
```

Comment:
Suggestion from Anh: Add diagnostic event for PreprovisionedVMType and PreprovisionedVm values.